packages/server/src/login/login-course.service.ts

```diff
@@ -31,38 +31,36 @@ export class LoginCourseService {
     const userCourses = [];
 
     for (const c of info.courses) {
-      const course: CourseModel = await this.courseSectionToCourse(
-        c.course,
-        c.section,
-      );
-
-      if (course) {
-        const userCourse = await this.courseToUserCourse(
-          user.id,
-          course.id,
-          Role.STUDENT,
-        );
-        userCourses.push(userCourse);
-      }
-    }
+      if (isKhouryCourse(c)) {
+        const course = await this.courseCRNToCourse(c.crn, c.semester);
 
-    if (info.ta_courses) {
-      for (const c of info.ta_courses) {
-        // Query for all the courses which match the name of the generic course from Khoury
-        const courseMappings = (
-          await CourseSectionMappingModel.find({
-            where: { genericCourseName: c.course }, // TODO: Add semester support
-            relations: ['course'],
-          })
-        ).filter((cm) => cm.course.enabled);
-
-        for (const courseMapping of courseMappings) {
-          const taCourse = await this.courseToUserCourse(
+        if (course) {
+          const userCourse = await this.courseToUserCourse(
             user.id,
-            courseMapping.courseId,
-            c.instructor === 1 ? Role.PROFESSOR : Role.TA,
+            course.id,
+            this.convertKhouryRole(c.role),
           );
-          userCourses.push(taCourse);
+          userCourses.push(userCourse);
+        }
+      } else {
+        // c is a KhouryProfCourse
+        // only need to inspect one of the CRNs from the list
+        // b/c they should all map to the same section group
```

Comment:
This assumption is not necessarily true. An instructor can be part of multiple section groups as they can be teaching multiple different courses where each course belongs to a different section group so we would have to iterate through all the courses

Reply:
This is analyzing a single section group when iterating through all the prof's courses. So when trying to check if one of the professor's section groups is registered for the app, I think it should be fine to check one CRN inside of that. Either all or zero CRNs will be registered inside a Khoury professor course

Reply:
Ohhh my bad yep all good!